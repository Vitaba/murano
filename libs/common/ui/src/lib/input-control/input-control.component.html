<div [class]="styles.container">
    <input [class]="styles.input" [type]="config.type">
  </div>