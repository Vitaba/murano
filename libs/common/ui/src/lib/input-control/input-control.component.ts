// tslint:disable: no-unsafe-any no-any
import { AfterViewInit, ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnChanges, SimpleChanges } from '@angular/core';
import { FormControl } from '@angular/forms';
import { GuardError, InputControlConfig, InputControlStyles, isInputControlConfig, isInputControlStyles } from '@vitaba/common-utils';
import { BehaviorSubject } from 'rxjs';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  selector: 'vitaba-input-control',
  styleUrls: ['./input-control.component.scss'],
  templateUrl: './input-control.component.html',
})
export class InputControlComponent implements OnChanges, AfterViewInit {
  public changes!: BehaviorSubject<SimpleChanges>;
  public errors!: GuardError;

  @Input() public config: InputControlConfig = {
    fControl: new FormControl('sample'),
    placeholder: 'placeholder',
    type: 'text',
  };

  @Input() public styles: InputControlStyles = {
    container: `md:w-2/3`,
    // tslint:disable-next-line:ter-max-len
    input: `bg-gray-200 appearance-none border-2 border-gray-200 rounded w-auto py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-purple-500`,
  };

  public constructor(
    private readonly _changeDetectorRef: ChangeDetectorRef,
  ) {}

  public ngOnChanges(changes: SimpleChanges): void {
    if (!this.changes) {
      this.changes = new BehaviorSubject(changes);
    }
    this.changes.next(changes);
  }

  public ngAfterViewInit(): void {
    if (!this.changes) {
      this.changes = new BehaviorSubject(null);
    }
    this.changes.subscribe((changes: SimpleChanges) => {
      if (changes.config) {
        const validation =
        this.validateInputControlConfig(
          changes.config.currentValue as InputControlConfig);

        if (validation) {
          this.errors = { ...this.errors, ...validation };
        }
      }

      if (changes.styles) {
        const validation =
        this.validateInputControlStyles(
          changes.styles.currentValue as InputControlStyles);

        if (validation) {
          this.errors = { ...this.errors, ...validation };
        }
      }
      this._changeDetectorRef.detectChanges();
    });
  }

  public validateInputControlConfig(value: InputControlConfig) {
    const validations = [{
      name: 'placeholder',
      type: 'string',
    },
      {
        name: 'type',
        type: 'string',
      },
    ];

    if (value.type === 'number' || value.type === 'range') {
      validations.push({
        name: 'min',
        type: 'number',
      });
      validations.push({
        name: 'max',
        type: 'number',
      });
      validations.push({
        name: 'step',
        type: 'number',
      });
    }

    const validation = isInputControlConfig(value, validations);

    return !validation.valid ? validation : undefined;
  }

  public validateInputControlStyles(value: InputControlStyles) {
    const validations = [
      {
        name: 'container',
        type: 'string',
      },
      {
        name: 'input',
        type: 'string',
      },
    ];

    const validation = isInputControlStyles(value, validations);

    return !validation.valid ? validation : undefined;
  }
}
