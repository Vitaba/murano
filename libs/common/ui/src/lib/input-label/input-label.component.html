<ng-container *ngIf="!errors; else showErrors">
<div [class]="styles.container">
    <label [class]="styles.label">
      <ng-container *ngIf="data.required">
      <span [class]="styles.required"> {{data.required}}</span>
      </ng-container>
      {{ data.label }}
    </label>
  </div>
  </ng-container>
  <ng-template #showErrors>
    {{ errors | json}}
  </ng-template>