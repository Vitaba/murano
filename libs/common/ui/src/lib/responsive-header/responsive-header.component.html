<div [class]="styles.container">

    <div [class]="styles.titleContainer">
      <ng-content select=".company-logo"></ng-content>
      <span [class]="styles.title">{{ data.title }}</span>
    </div>
    <div [class]="styles.formContainer">
      <div [class]="styles.formSectionOne">
        <form [formGroup]="form">

          <vitaba-input-control [config]="{
          type: data.formControlType,
          placeholder: data.formControlPlaceholder,
          fControl: form.get('search')
          }"
          [styles]="{
           container: styles.formControlContainer,
           input: styles.formControlInput
          }">
          <ng-container class="icon">
            <div class="pointer-events-none absolute inset-y-0 left-0 pl-4 flex items-center">
              <svg class="fill-current pointer-events-none text-gray-600 w-4 h-4" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path d="M12.9 14.32a8 8 0 1 1 1.41-1.41l5.35 5.33-1.42 1.42-5.33-5.34zM8 14A6 6 0 1 0 8 2a6 6 0 0 0 0 12z"></path></svg>
            </div>
          </ng-container>
          </vitaba-input-control>

        </form>
      </div>
      <div [class]="styles.formSectionTwo">
        <button [class]="styles.menuButton">
          <ng-template #menuDefaultTemplate>
            <svg class="fill-current h-3 w-3" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
              <title>Menu</title>
              <path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z" />
            </svg>
          </ng-template>
          <ng-container [ngTemplateOutlet]="menuExtraTemplate || menuDefaultTemplate">
          </ng-container>
         </button>
      </div>
  </div>