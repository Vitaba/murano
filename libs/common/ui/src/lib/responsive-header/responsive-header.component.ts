// tslint:disable: no-unsafe-any no-any
import { AfterViewInit, ChangeDetectionStrategy, ChangeDetectorRef, Component, ContentChild, EventEmitter, Input, OnChanges, Output, SimpleChanges, TemplateRef } from '@angular/core';
import { FormBuilder } from '@angular/forms';
import { GuardError, isResponsiveHeaderData, isResponsiveHeaderStyles, ResponsiveHeaderData, ResponsiveHeaderStyles } from '@vitaba/common-utils';
import { BehaviorSubject } from 'rxjs';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  selector: 'vitaba-responsive-header',
  styleUrls: ['./responsive-header.component.scss'],
  templateUrl: './responsive-header.component.html',
})
export class ResponsiveHeaderComponent implements AfterViewInit, OnChanges {
  public changes!: BehaviorSubject<SimpleChanges>;
  public errors!: GuardError;

  @Input() public data: ResponsiveHeaderData = {
    formControlPlaceholder: `Search the items "(Press / to focus)"`,
    formControlType: 'text',
    title: 'Murano Admin',
  };

  @Input() public styles: ResponsiveHeaderStyles = {
    container: 'flex bg-white-400 border-b border-gray-300',
    formContainer: 'flex w-5/6 p-2',
    formControlContainer: 'flex',
    // tslint:disable-next-line:ter-max-len
    formControlInput: 'transition focus:outline-0 border border-transparent focus:bg-white focus:border-gray-300 placeholder-gray-600 rounded-lg bg-gray-200 py-2 pr-4 pl-10 block w-full appearance-none leading-normal ds-input outline-none',
    formSectionOne: 'w-11/12',
    formSectionTwo: 'w-1/12 flex items-center justify-center',
    // tslint:disable-next-line:ter-max-len
    menuButton: `flex items-center px-3 py-3 border rounded text-gray-500 outline-none focus:outline-none`,
    title: 'font-sans text-xl tracking-tight text-black-400 select-none',
    titleContainer: 'flex items-center justify-center w-1/6 p-2',
  };

  @ContentChild('menuExtraTemplate', { static: false })

  @Output() public readonly changed = new EventEmitter<any>();

  public menuExtraTemplate!: TemplateRef<any>;

  public form = this._fBuilder.group({
    search: '',
  });

  public constructor(
    private readonly _fBuilder: FormBuilder,
    private readonly _changeDetectorRef: ChangeDetectorRef) {}

  public ngOnChanges(changes: SimpleChanges): void {
    if (!this.changes) {
      this.changes = new BehaviorSubject(changes);
    }
    this.changes.next(changes);
  }

  public ngAfterViewInit(): void {
    this.changes.subscribe((changes: SimpleChanges) => {
      if (changes.data) {
        const validation =
        this.validateResponsiveHeaderData(
          changes.data.currentValue as ResponsiveHeaderData);

        if (validation) {
          this.errors = { ...this.errors, ...validation };
        }
      }
      if (changes.styles) {
        const validation =
        this.validateHeaderStyles(
          changes.styles.currentValue as ResponsiveHeaderStyles);

        if (validation) {
          this.errors = { ...this.errors, ...validation };
        }
      }
      this._changeDetectorRef.detectChanges();
      this.listenFormChanges();
    });
  }

  public listenFormChanges() {
    this.form.valueChanges.subscribe(() => {
      this.changed.emit(this.form.value);
    });
  }

  public validateResponsiveHeaderData(value: ResponsiveHeaderData) {
    const validations = [
      {
        name: 'formControlPlaceholder',
        type: 'string',
      },
      {
        name: 'formControlType',
        type: 'string',
      },
      {
        name: 'title',
        type: 'string',
      },
    ];

    const validation = isResponsiveHeaderData(value, validations);

    return !validation.valid ? validation : undefined;
  }

  public validateHeaderStyles(value: ResponsiveHeaderStyles) {
    const validations = [
      {
        name: 'container',
        type: 'string',
      },
      {
        name: 'formContainer',
        type: 'string',
      },
      {
        name: 'formControlContainer',
        type: 'string',
      },
      {
        name: 'formSectionOne',
        type: 'string',
      },
      {
        name: 'formSectionTwo',
        type: 'string',
      },
      {
        name: 'menuButton',
        type: 'string',
      },
      {
        name: 'title',
        type: 'string',
      },
      {
        name: 'titleContainer',
        type: 'string',
      },
    ];

    const validation = isResponsiveHeaderStyles(value, validations);

    return !validation.valid ? validation : undefined;
  }
}
