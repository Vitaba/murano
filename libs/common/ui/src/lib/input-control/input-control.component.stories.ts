// tslint:disable:no-default-export object-literal-sort-keys ter-max-len

import { FormControl } from '@angular/forms';
import { text } from '@storybook/addon-knobs';
import { InputControlComponent } from './input-control.component';

export default {
  title: 'Input Control',
};

const config = {
  fControl: new FormControl('sample'),
  placeholder: 'placeholder',
  type: 'text',
};

const styles = {
  container: `md:w-2/3`,
  input: `bg-gray-200 appearance-none border-2 border-gray-200 rounded w-auto py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-purple-500`,
};

export const basic = () => ({
  moduleMetadata: {
    declarations: [InputControlComponent],
    imports: [],
  },
  props: {
    config_placeholder: text('placeholder', config.placeholder, 'config'),
    config_type: text('type', config.type, 'config'),
    styles_container: text('container', styles.container, 'styles'),
    styles_input: text('input', styles.input, 'styles'),
  },
  template: `
  <vitaba-input-control
    [config]="{
    placeholder: config_placeholder,
    type: config_type
    }"
    [styles]="{
    container: styles_container,
    input: styles_input
    }"
    > </vitaba-input-control>
  `,
});
