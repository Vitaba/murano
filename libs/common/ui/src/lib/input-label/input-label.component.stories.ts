// tslint:disable:no-default-export object-literal-sort-keys ter-max-len

import { text } from '@storybook/addon-knobs';
import { InputLabelComponent } from './input-label.component';

export default {
  title: 'Input Label',
};

const data = {
  label: 'Name',
};

const styles = {
  container: `md:w-1/3`,
  label: `block text-gray-500 font-bold md:text-right mb-1 md:mb-0 pr-4`,
};

export const basic = () => ({
  moduleMetadata: {
    declarations: [InputLabelComponent],
    imports: [],
  },
  props: {
    data_label: text('label', data.label, 'data'),
    styles_container: text('container', styles.container, 'styles'),
    styles_label: text('label', styles.label, 'styles'),
  },
  template: `
  <vitaba-input-label
    [data]="{
    label: data_label
    }"
    [styles]="{
    container: styles_container,
    label: styles_label
    }"
    > </vitaba-input-label>
  `,
  stylesUrl: [
    'node_modules/tailwindcss/dist/base',
    'node_modules/tailwindcss/dist/components',
    'node_modules/tailwindcss/dist/utilities',
  ]
});
