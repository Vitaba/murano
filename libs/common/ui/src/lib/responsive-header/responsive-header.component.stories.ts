// tslint:disable:no-default-export object-literal-sort-keys ter-max-len

import { ReactiveFormsModule } from '@angular/forms';
import { text } from '@storybook/addon-knobs';
import { InputControlComponent } from '../input-control/input-control.component';
import { ResponsiveHeaderComponent } from './responsive-header.component';

export default {
  title: 'Responsive Header',
};

const data = {
  formControlPlaceholder: `Search the items (Press / to focus)`,
  formControlType: 'text',
  title: 'Murano Admin',
};

const styles = {
  container: 'flex bg-white-400 border-b border-gray-300',
  formContainer: 'flex w-5/6 p-2',
  formControlContainer: 'flex',
    // tslint:disable-next-line:ter-max-len
  formControlInput: 'transition focus:outline-0 border border-transparent focus:bg-white focus:border-gray-300 placeholder-gray-600 rounded-lg bg-gray-200 py-2 pr-4 pl-10 block w-full appearance-none leading-normal ds-input outline-none',
  formSectionOne: 'w-11/12',
  formSectionTwo: 'w-1/12 flex items-center justify-center',
    // tslint:disable-next-line:ter-max-len
  menuButton: `flex items-center px-3 py-3 border rounded text-gray-500 outline-none focus:outline-none`,
  title: 'font-sans text-xl tracking-tight text-black-400 select-none',
  titleContainer: 'flex items-center justify-center w-1/6 p-2',
};

export const basic = () => ({
  moduleMetadata: {
    declarations: [ResponsiveHeaderComponent, InputControlComponent],
    imports: [ReactiveFormsModule],
  },
  props: {
    data_formControlPlaceholder: text('formControlPlaceholder', data.formControlPlaceholder, 'data'),
    data_formControlType: text('formControlType', data.formControlType, 'data'),
    data_title: text('title', data.title, 'data'),
    styles_container: text('container', styles.container, 'styles'),
    styles_formContainer: text('formContainer', styles.formContainer, 'styles'),
    styles_formControlContainer: text('formControlContainer', styles.formControlContainer, 'styles'),
    styles_formControlInput: text('formControlInput', styles.formControlInput, 'styles'),
    styles_formSectionOne: text('formSectionOne', styles.formSectionOne, 'styles'),
    styles_formSectionTwo: text('formSectionTwo', styles.formSectionTwo, 'styles'),
    styles_menuButton: text('menuButton', styles.menuButton, 'styles'),
    styles_title: text('title', styles.title, 'styles'),
    styles_titleContainer: text('titleContainer', styles.titleContainer, 'styles'),
  },
  template: `
  <vitaba-responsive-header
    [data]="{
      formControlPlaceholder: data_formControlPlaceholder,
      formControlType: data_formControlType,
      title: data_title
    }"
    [styles]="{
    container: styles_container,
    formContainer: styles_formContainer,
    formControlContainer: styles_formControlContainer,
    formControlInput: styles_formControlInput,
    formSectionOne: styles_formSectionOne,
    formSectionTwo: styles_formSectionTwo,
    menuButton: styles_menuButton,
    title: styles_title,
    titleContainer: styles_titleContainer
    }"
    > </vitaba-responsive-header>
  `,
  stylesUrl: [
    'node_modules/tailwindcss/dist/base',
    'node_modules/tailwindcss/dist/components',
    'node_modules/tailwindcss/dist/utilities',
  ],
});
