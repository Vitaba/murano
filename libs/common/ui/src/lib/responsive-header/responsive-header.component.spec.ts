
// tslint:disable: no-any component-max-inline-declarations ter-max-len
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { InputControlComponent } from '../input-control/input-control.component';
import { InputLabelComponent } from '../input-label/input-label.component';
import { ResponsiveHeaderComponent } from './responsive-header.component';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  selector: 'vitaba-responsive-header-sample-implementation',
  template: `
    <vitaba-responsive-header [data]="data" [styles]="styles">
    </vitaba-responsive-header>
  `,
})
class TestHostComponent {
  public data = {
    formControlPlaceholder: `Search the items "(Press / to focus)"`,
    formControlType: 'text',
    title: 'Murano Admin',
  };

  public styles = {
    container: 'flex bg-white-400 border-b border-gray-300',
    formContainer: 'flex w-5/6 p-2',
    formControlContainer: 'flex',
    // tslint:disable-next-line:ter-max-len
    formControlInput: 'transition focus:outline-0 border border-transparent focus:bg-white focus:border-gray-300 placeholder-gray-600 rounded-lg bg-gray-200 py-2 pr-4 pl-10 block w-full appearance-none leading-normal ds-input outline-none',
    formSectionOne: 'w-11/12',
    formSectionTwo: 'w-1/12 flex items-center justify-center',
    // tslint:disable-next-line:ter-max-len
    menuButton: `flex items-center px-3 py-3 border rounded text-gray-500 outline-none focus:outline-none`,
    title: 'font-sans text-xl tracking-tight text-black-400 select-none',
    titleContainer: 'flex items-center justify-center w-1/6 p-2',
  };
}

describe('ResponsiveHeaderComponent', () => {
  let component: TestHostComponent;
  let fixture: ComponentFixture<TestHostComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [ReactiveFormsModule],
      declarations: [TestHostComponent, ResponsiveHeaderComponent, InputLabelComponent, InputControlComponent],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TestHostComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    fixture.detectChanges();
    expect(component).toBeTruthy();
  });

  it('should be default', () => {
    fixture.changeDetectorRef.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should throw errors when data is not in right format', () => {
    component.data = {
      title: 1,
    } as any;
    fixture.changeDetectorRef.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
