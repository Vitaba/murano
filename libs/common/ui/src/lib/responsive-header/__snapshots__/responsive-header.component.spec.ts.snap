// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveHeaderComponent should be default 1`] = `
<vitaba-responsive-header-sample-implementation
  data={[Function Object]}
  styles={[Function Object]}
>
  <vitaba-responsive-header
    ng-reflect-data="[object Object]"
    ng-reflect-styles="[object Object]"
  >
    <div
      class="flex bg-white-400 border-b border-gray-300"
    >
      <div
        class="flex items-center justify-center w-1/6 p-2"
      >
        <span
          class="font-sans text-xl tracking-tight text-black-400 select-none"
        >
          Murano Admin
        </span>
      </div>
      <div
        class="flex w-5/6 p-2"
      >
        <div
          class="w-11/12"
        >
          <form
            class="ng-untouched ng-pristine ng-valid"
            ng-reflect-form="[object Object]"
            novalidate=""
          >
            <vitaba-input-control
              ng-reflect-config="[object Object]"
              ng-reflect-styles="[object Object]"
            >
              
              
              <div
                class="flex"
              >
                <div
                  class="relative"
                >
                  
                  <div
                    class="pointer-events-none absolute inset-y-0 left-0 pl-4 flex items-center"
                  >
                    <svg
                      class="fill-current pointer-events-none text-gray-600 w-4 h-4"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12.9 14.32a8 8 0 1 1 1.41-1.41l5.35 5.33-1.42 1.42-5.33-5.34zM8 14A6 6 0 1 0 8 2a6 6 0 0 0 0 12z"
                      />
                    </svg>
                  </div>
                </div>
                <input
                  class="transition focus:outline-0 border border-transparent focus:bg-white focus:border-gray-300 placeholder-gray-600 rounded-lg bg-gray-200 py-2 pr-4 pl-10 block w-full appearance-none leading-normal ds-input outline-none ng-untouched ng-pristine ng-valid"
                  ng-reflect-form="[object Object]"
                  placeholder="Search the items \\"(Press / to focus)\\""
                  type="text"
                />
              </div>
              
            </vitaba-input-control>
          </form>
        </div>
        <div
          class="w-1/12 flex items-center justify-center"
        >
          <button
            class="flex items-center px-3 py-3 border rounded text-gray-500 outline-none focus:outline-none"
          >
            
            
            <svg
              class="fill-current h-3 w-3"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Menu
              </title>
              <path
                d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </vitaba-responsive-header>
</vitaba-responsive-header-sample-implementation>
`;

exports[`ResponsiveHeaderComponent should throw errors when data is not in right format 1`] = `
<vitaba-responsive-header-sample-implementation
  data={[Function Object]}
  styles={[Function Object]}
>
  <vitaba-responsive-header
    ng-reflect-data="[object Object]"
    ng-reflect-styles="[object Object]"
  >
    <div
      class="flex bg-white-400 border-b border-gray-300"
    >
      <div
        class="flex items-center justify-center w-1/6 p-2"
      >
        <span
          class="font-sans text-xl tracking-tight text-black-400 select-none"
        >
          1
        </span>
      </div>
      <div
        class="flex w-5/6 p-2"
      >
        <div
          class="w-11/12"
        >
          <form
            class="ng-untouched ng-pristine ng-valid"
            ng-reflect-form="[object Object]"
            novalidate=""
          >
            <vitaba-input-control
              ng-reflect-config="[object Object]"
              ng-reflect-styles="[object Object]"
            >
              
               
              
            </vitaba-input-control>
          </form>
        </div>
        <div
          class="w-1/12 flex items-center justify-center"
        >
          <button
            class="flex items-center px-3 py-3 border rounded text-gray-500 outline-none focus:outline-none"
          >
            
            
            <svg
              class="fill-current h-3 w-3"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Menu
              </title>
              <path
                d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </vitaba-responsive-header>
</vitaba-responsive-header-sample-implementation>
`;
